docs: use quantinuum-sphinx theme

daniel-mills-cqc opened the pull request:
I've created this new branch with your changes on top of main. Let me know if this is actually helpful.

daniel-mills-cqc commented on .github/workflows/deploy_docs.yml:
I think that the path to the artifact to be uploaded will need to be changed to he build directory.

CalMacCQ replied:
hopefully fixed in [44e823d](https://github.com/CQCL/Qermit/pull/120/commits/44e823d67fcb41aa8d2b5b70dce06875aed65d8b)

CalMacCQ commented:
thanks for sorting this out, I should be able to finish this off today

CalMacCQ commented on pyproject.toml:
```diff
@@ -22,10 +22,10 @@ pytest = {version="^8.1.1", optional=true}
 qiskit-ibm-provider = {version="^0.10.0", optional=true}
 pytket-quantinuum = {version="0.40.0", optional=true}
 ruff = {version="^0.7.4", optional=true}
+furo = {version="^2024.8.6", optional=true}
```
Would you prefer this under "extras"?

CalMacCQ replied:
Done, let me know if I got it right

daniel-mills-cqc replied:
Looks good to me! Thanks!

CalMacCQ commented on docs/_static/nav-config.js:
```diff
@@ -0,0 +1,20 @@
+        },
+        {
+            "title": "Manual",
+            "href": "manual/manual_index.html",
```
By the way these links are configurable (at least with this version of quantinuum-sphinx) 

feel free to suggest other places for the navbar to go

daniel-mills-cqc reviewed (commented):
Looking great!
Would you mind checking the warning from the build. I see a warning
```
WARNING: html_static_path entry 'quantinuum-sphinx/_static' does not exist
WARNING: favicon file 'quantinuum-sphinx/_static/assets/quantinuum_favicon.svg' does not exist
```
Could those be fixed?

CalMacCQ commented:
> Looking great! Would you mind checking the warning from the build. I see a warning
> 
> ```
> WARNING: html_static_path entry 'quantinuum-sphinx/_static' does not exist
> WARNING: favicon file 'quantinuum-sphinx/_static/assets/quantinuum_favicon.svg' does not exist
> ```
> 
> Could those be fixed?

Ah thanks, I can fix that

CalMacCQ commented:
> Looking great! Would you mind checking the warning from the build. I see a warning
> 
> ```
> WARNING: html_static_path entry 'quantinuum-sphinx/_static' does not exist
> WARNING: favicon file 'quantinuum-sphinx/_static/assets/quantinuum_favicon.svg' does not exist
> ```
> 
> Could those be fixed?

hmm I'm not getting those.. Perhaps a submodule issue?

Try doing

```shell
git submodule update --init --recursive
```

Also make sure you're on commit `a8ac27e089824099708da2cea9cef262e4df3ad2 quantinuum-sphinx (heads/dist)`

run this to check

```shell
git submodule status
```

CalMacCQ commented:
I've also added a README in the `docs` folder. Let me know if you're happy with this. 

Would you recommend using poetry or pip in local build and github worklfows?

CalMacCQ commented:
> Could those be fixed?

Warnings now fixed, hadn't checked out the submodule in C.I. properly.

These warnings were actually quite serious. I've added a `-W` flag to fail the sphinx C.I. build on warnings to catch this sort of thing in future.

CalMacCQ commented on pyproject.toml:
```diff
@@ -18,18 +18,25 @@ python = ">=3.10, <3.13"
-tests = ["pytest", "mypy", "ruff", "pytest-cov", "qiskit-ibm-provider", "pytket-quantinuum"]
-docs = ["sphinx", "sphinx-book-theme", "pytket-quantinuum"]
+tests = [
+    "pytest",
```
think my linter made these changes?

daniel-mills-cqc commented:
> I've also added a README in the `docs` folder. Let me know if you're happy with this.
> 
> Would you recommend using poetry or pip in local build and github workflows?

Thanks, looking good!

I don't really recommend using either in particular, but I'm happy to stick with pip in this readme.

daniel-mills-cqc commented on .github/workflows/build_and_test.yml:
```diff
@@ -33,20 +33,19 @@ jobs:
       - name: Save documentation
         uses: actions/upload-artifact@v4
         with:
-          name: docs_html
```
I don't think it matters hugely, but I think only the build itself really needs uploading here. Would you mind making that change?

CalMacCQ replied:
Fixed now

daniel-mills-cqc commented:
> > Could those be fixed?
> 
> Warnings now fixed, hadn't checked out the submodule in C.I. properly.
> 
> These warnings were actually quite serious. I've added a `-W` flag to fail the sphinx C.I. build on warnings to catch this sort of thing in future.

Locally I am still getting
```
html_static_path entry 'quantinuum-sphinx/_static' does not exist
```
Does that surprise you?
I think I need to clone the quantinuum-sphinx subodule. If I clone -> https://github.com/CQCL/quantinuum-sphinx/tree/a8ac27e089824099708da2cea9cef262e4df3ad2 into the `quantinuum-sphinx` directory will that do the job?

daniel-mills-cqc commented on docs/README.md:
````diff
@@ -0,0 +1,31 @@
+
+2. Install Qermit and docs dependencies
+
+```shell
+python -m venv ".venv"
+source .venv/bin/activate
+pip install -e .[docs] 
+```
````
I think we could remove the reference to venv and just allow the user to use whichever means they prefer.
You could also mention at this point that poetry is supported.

daniel-mills-cqc reviewed (commented):
Just this last small comment, then it's good to go!
